### ISSUE TITLE
Geofencing sample:  "Geofence service is not available now"

### ISSUE THREAD

**Issue by RikdeBoer** [googlesamples/android-play-location#161]
Using API 26 / 27

App compiles and starts fine.
App pops up in emulator (set to Oreo 8 device), showing two buttons "Add Geofences" and "Remove Geofences" (greyed-out).

Press "Add Geofences".

Seeing this in the log: 
com.google.android.gms.location.sample.geofencing W/MainActivity: Geofence service is not available now

Any special Settings that I'm not aware of?

**Comment by RikdeBoer** [comment]
Yes. Didn't realise I had to do this (at least in emulator for Oreo)

**Settings -> Security & Location ->  Location -> App-level Permissions -> Location Updates Foreground ON**

**Comment by RikdeBoer** [comment]
After making the above setting, I cannot get this app to do anything noticeable, at least on Oreo 8.

There's a button "Add Geofences". When I look in the debugger I can see it adds to locations, SFO and GOOGLE with a 1 mile geofence radius.

Using the 3 dots command on the emulator I have set lat/lon to be close then far away from SFO and GOOGLE. Nothing seems to get triggered. 

Some instruction on how to use this app and what it is intended to do would be most welcome...

**Comment by ergun-p** [comment]
+1

**Comment by rohitsharma181589** [comment]
I am facing this issue on testing device 'LG Q6' 
Android version 7.1.1
I am using the sample app for geofencing and when i click on Add Geofence button it throws an error 
"Geofence service is not available now" ,please suggest on this.

**Comment by nandovelazquez** [comment]
I faced the same problem and I fix it restarting the device.
I'm using a Pixel with android 9.

Also you can use an app called _Lockito_ to fake a gps route so the Geofacing app can determine if it enters/exit of the geofence zone.

**Comment by olddustysocksunderthecouch** [comment]
On API 25 I had to go into Settings --> Location --> Mode --> select either Battery saving or High Accuracy. Geofences use wifi and cellular networks to determine your location so if you leave it in device mode which only has access to GPS, geofences won't work.
